server/src/main/java/org/elasticsearch/action/index/IndexRequestBuilder.java

```diff
@@ -202,31 +248,145 @@ public IndexRequestBuilder setIfSeqNo(long seqNo) {
      * {@link org.elasticsearch.index.engine.VersionConflictEngineException} will be thrown.
      */
     public IndexRequestBuilder setIfPrimaryTerm(long term) {
-        request.setIfPrimaryTerm(term);
+        this.ifPrimaryTerm = term;
         return this;
     }
 
     /**
      * Sets the ingest pipeline to be executed before indexing the document
      */
     public IndexRequestBuilder setPipeline(String pipeline) {
-        request.setPipeline(pipeline);
+        this.pipeline = pipeline;
         return this;
     }
 
     /**
      * Sets the require_alias flag
      */
     public IndexRequestBuilder setRequireAlias(boolean requireAlias) {
-        request.setRequireAlias(requireAlias);
+        this.requireAlias = requireAlias;
         return this;
     }
 
     /**
      * Sets the require_data_stream flag
      */
     public IndexRequestBuilder setRequireDataStream(boolean requireDataStream) {
-        request.setRequireDataStream(requireDataStream);
+        this.requireDataStream = requireDataStream;
+        return this;
+    }
+
+    public IndexRequestBuilder setRefreshPolicy(WriteRequest.RefreshPolicy refreshPolicy) {
+        this.refreshPolicy = refreshPolicy;
         return this;
     }
+
+    public IndexRequestBuilder setRefreshPolicy(String refreshPolicy) {
+        this.refreshPolicyString = refreshPolicy;
+        return this;
+    }
+
+    @Override
+    public IndexRequest request() {
+        validate();
+        IndexRequest request = new IndexRequest();
+        super.apply(request);
+        request.id(id);
+        if (sourceMap != null) {
+            if (sourceContentType == null) {
```

Comment:
I think these might scan better if you insisted on `!= null` checks throughout. Then for example this top one would become:

```
if (sourceMap != null && sourceContentType != null) {
    request.source(sourceMap, sourceContentType);
else if (sourceContentType != null) {
    request.source(sourceMap);
}
```